### PR TITLE
add getSupportedOption override to include mapping file path constant

### PR DESCRIPTION
**Pull request by wangwh03**
https://github.com/gradle/incap/issues/54

### CODE REVIEW

File: ap-client/src/test/java/org/gradle/incap/BaseIncrementalAnnotationProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+public class BaseIncrementalAnnotationProcessorTest {
+    private static final String ADDITIONAL_OPTION = "option";
+
+    private BaseIncrementalAnnotationProcessor incrementalAnnotationProcessorUnderTest;
+    private IncrementalFiler incrementalFilerForTest;
+
+    @Before
+    public void setUp() {
+        incrementalFilerForTest = EasyMock.createMock(IncrementalFiler.class);
```
Comment (stephanenicolas) [inline ap-client/src/test/java/org/gradle/incap/BaseIncrementalAnnotationProcessorTest.java]: static improt

Comment (stephanenicolas) [inline ap-client/src/test/java/org/gradle/incap/BaseIncrementalAnnotationProcessorTest.java]: mockIncrementalFiler

File: ap-client/src/test/java/org/gradle/incap/BaseIncrementalAnnotationProcessorTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        incrementalAnnotationProcessorUnderTest = new BaseIncrementalAnnotationProcessor() {
+            @Override
+            public synchronized void init(ProcessingEnvironment processingEnv) {
+                this.incrementalFiler = incrementalFilerForTest;
+            }
+
+            @Override public Set<String> getAdditionalSupportedOptions() {
+                return Collections.singleton(ADDITIONAL_OPTION);
```
Comment (stephanenicolas) [inline ap-client/src/test/java/org/gradle/incap/BaseIncrementalAnnotationProcessorTest.java]: static import

File: ap-client/src/test/java/org/gradle/incap/BaseIncrementalAnnotationProcessorTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+public class BaseIncrementalAnnotationProcessorTest {
+    private static final String ADDITIONAL_OPTION = "option";
+
+    private BaseIncrementalAnnotationProcessor incrementalAnnotationProcessorUnderTest;
+    private IncrementalFiler incrementalFilerForTest;
+
+    @Before
+    public void setUp() {
+        incrementalFilerForTest = EasyMock.createMock(IncrementalFiler.class);
+        incrementalAnnotationProcessorUnderTest = new BaseIncrementalAnnotationProcessor() {
```
Comment (stephanenicolas) [inline ap-client/src/test/java/org/gradle/incap/BaseIncrementalAnnotationProcessorTest.java]: You can create a root class for this one in the test src tree

Comment (stephanenicolas) [inline ap-client/src/test/java/org/gradle/incap/BaseIncrementalAnnotationProcessorTest.java]: static class

File: ap-client/src/test/java/org/gradle/incap/BaseIncrementalAnnotationProcessorTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            @Override
+            public synchronized void init(ProcessingEnvironment processingEnv) {
```
Comment (stephanenicolas) [inline ap-client/src/test/java/org/gradle/incap/BaseIncrementalAnnotationProcessorTest.java]: init is not used.

Comment (stephanenicolas) [inline ap-client/src/test/java/org/gradle/incap/BaseIncrementalAnnotationProcessorTest.java]: you can use the constructor of you AP class.

**Comment by wangwh03** [comment]
Review comments incorporated.

**Pull request merged**